overlord/snapstate/snapmgr.go

```diff
@@ -428,6 +458,17 @@ func (m *SnapManager) doMountSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// record the try mode in the state
+	t.State().Lock()
+	t.Set("old-trymode", snapst.TryMode())
+	if ss.TryMode() {
+		snapst.Flags |= TryMode
+	} else {
+		snapst.Flags &= ^TryMode
+	}
+	Set(t.State(), ss.Name, snapst)
+	t.State().Unlock()
```

Comment:
No, definitely not. It should probably follow the pattern we've used (or tried to?) in the other handlers: persist such changes at the end of the handler, so errors abort them properly.